File: src/main/java/com/perfscope/ui/DatabaseView.java
package com.perfscope.ui;

import com.perfscope.model.CommData;

import javafx.scene.control.*;
import javafx.scene.layout.BorderPane;
import javafx.scene.layout.Priority;
import javafx.scene.layout.VBox;
import org.jooq.Record3;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class DatabaseView {
    
    private static final Logger logger = LoggerFactory.getLogger(DatabaseView.class);

    public BorderPane createCommView(String databasePath, CommData commData) {
        BorderPane tabContent = new BorderPane();
        
        ListView<String> threadListView = new ListView<>();
        
        for (Record3<Long, Integer, Integer> thread : commData.getThreads()) {
            threadListView.getItems().add(String.format("PID: %d, TID: %d", thread.value2(), thread.value3()));
        }
        
        CallTreeView callTreeView = new CallTreeView();
        
        threadListView.getSelectionModel().selectedItemProperty().addListener((observable, oldValue, newValue) -> {
            if (newValue != null) {
                int selectedIndex = threadListView.getSelectionModel().getSelectedIndex();
                if (selectedIndex >= 0 && selectedIndex < commData.getThreads().size()) {
                    Record3<Long, Integer, Integer> selectedThread = commData.getThreads().get(selectedIndex);
                    Long threadId = selectedThread.value1();
                    
                    callTreeView.loadThreadData(databasePath, commData.getComm().getId().longValue(), threadId);
                }
            }
        });
        
        SplitPane splitPane = new SplitPane();

        VBox threadBox = new VBox();
        Label threadsLabel = new Label("Threads");
        threadsLabel.setStyle("-fx-font-weight: bold; -fx-padding: 5px;");
        threadBox.getChildren().addAll(threadsLabel, threadListView);
        VBox.setVgrow(threadListView, Priority.ALWAYS);
        
        splitPane.getItems().addAll(threadBox, callTreeView);
        splitPane.setDividerPositions(0.2);

        threadBox.setMinWidth(100);
        threadBox.setMaxWidth(300);

        tabContent.setCenter(splitPane);
        
        return tabContent;
    }
} 

File: src/main/kotlin/com/perfscope/view/DatabaseView.kt
package com.perfscope.view

import com.perfscope.model.Command
import com.perfscope.model.Thread
import javafx.scene.layout.BorderPane
import javafx.scene.control.Label
import javafx.scene.control.SplitPane
import javafx.scene.layout.VBox
import javafx.scene.control.ListView

class DatabaseView {

    fun createCommView(dbPath: String, command: Command): BorderPane {
        val tabContent = BorderPane()

        val threadListView = ListView<String?>()

        for (thread in command.threads) {
            threadListView.items += String.format("PID: %d, TID: %d", thread.pid, thread.tid)
        }

        val callTreeView = CallTreeView(dbPath)

        threadListView.selectionModel.selectedItemProperty().addListener { observable, oldValue, newValue ->
            if (newValue != null) {
                val selectedIndex: Int = threadListView.getSelectionModel().selectedIndex
                if (selectedIndex >= 0 && selectedIndex < command.threads!!.size) {
                    val selectedThread: Thread = command.threads[selectedIndex]
                    callTreeView.switchThread(command.id, selectedThread.id)
                }
            }
        }

        val threadsLabel = Label("Threads")
        threadsLabel.style = "-fx-font-weight: bold; -fx-padding: 5px;"

        val threadBox = VBox()
        threadBox.children.addAll(threadsLabel, threadListView)
        VBox.setVgrow(threadListView, javafx.scene.layout.Priority.ALWAYS)

        threadBox.minWidth = 100.0
        threadBox.maxWidth = 300.0

        val splitPane = SplitPane()
        splitPane.items.addAll(threadBox, callTreeView)
        splitPane.setDividerPositions(0.2)

        tabContent.center = splitPane
        return tabContent
    }
}